### DISCUSSION TITLE
Multiple parametrization of a fixture?

### DISCUSSION THREAD

**Discussion by sadra-barikbin** [pytest-dev/pytest#10534] [Q&A] [answered]
Is there a way to parametrize a fixture like we do by stacking `pytest.mark.parametrize`s on a test function?

**Comment by portyu9** [comment, marked as answer]
Hi @sadra-barikbin 

You don't stack multiple `@pytest.fixture(params=...)` decorators in the same way that you can stack `@pytest.mark.parametrize`.

If you want a Cartesian product, one clean approach is to make the parameters separate fixtures:

```python
@pytest.fixture(params=[1, 2])
def number(request):
    return request.param


@pytest.fixture(params=["a", "b"])
def letter(request):
    return request.param


@pytest.fixture
def combined(number, letter):
    return number, letter
```

Any test using `combined` will run with:

```text
(1, "a")
(1, "b")
(2, "a")
(2, "b")
```

Alternatively, if the pair conceptually belongs to one fixture:

```python
import itertools
import pytest

@pytest.fixture(params=list(itertools.product([1, 2], ["a", "b"])))
def combined(request):
    return request.param
```

Both produce the Cartesian-product behavior; the first form is usually nicer when the individual parameters are useful as fixtures themselves.

Please mark this answer as accepted if it helped, thank you.
